src/ng/http.js

```diff
@@ -1111,17 +1111,25 @@ function $HttpProvider() {
       /**
        * Resolves the raw $http promise.
        */
-      function resolvePromise(response, status, headers, statusText) {
+      function resolvePromise(response, status, headers, statusText, timedOut) {
         // normalize internal statuses to 0
         status = Math.max(status, 0);
 
-        (isSuccess(status) ? deferred.resolve : deferred.reject)({
+        var result = {
           data: response,
           status: status,
           headers: headersGetter(headers),
           config: config,
           statusText: statusText
```

Comment:
I didn't want to alter the current success result, since it logically can never be a timeout (hence `timeOut: false` would be redundant; I think of `timeOut` as a possible outcome of an error result). But maybe it'd better to add `timeOut: [true | false]` every time.

Another question: would you go with `timedOut`, `timeOut`, or `timeout`?